front/src/applications/operationalStudies/components/Scenario/getSimulationResults.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  updateAllowancesSettings,
+  updateIsUpdating,
+  updateSelectedProjection,
+  updateSelectedTrainId,
+  updateSimulation,
+} from 'reducers/osrdsimulation/actions';
+import { setFailure } from 'reducers/main';
+import { store } from 'Store';
+import i18n from 'i18next';
+import {
+  SimulationReport,
+  TimetableWithSchedulesDetails,
+  osrdEditoastApi,
+} from 'common/api/osrdEditoastApi';
+import { Train } from 'reducers/osrdsimulation/types';
+import { ApiError } from 'common/api/emptyApi';
+import { SerializedError } from '@reduxjs/toolkit';
+
+/**
+ * Recover the time table for all the trains
+ */
+
+export default async function getSimulationResults(
+  timetable: TimetableWithSchedulesDetails,
+  selectedTrainId?: number
+) {
+  const { selectedProjection, allowancesSettings } = store.getState().osrdsimulation;
+  store.dispatch(updateIsUpdating(true));
+  const trainSchedulesIDs = timetable.train_schedule_summaries.map((train) => train.id);
+
+  if (trainSchedulesIDs && trainSchedulesIDs.length > 0) {
+    if (!selectedTrainId || trainSchedulesIDs.length === 1) {
```

Comment:
You should update the selectedTrainId (with the id of the first trainSchedule) if:
- selectedTrainId is undefined
- selectedTrainId is not in trainSchedulesIds

```suggestion
    if (!selectedTrainId || !trainSchedulesIDs.includes(selectedTrainId)) {
```

Reply:
Done :)